### ISSUE TITLE
Newbie. Read simple webserver is only 65 KB compiled.  Tried hello_world.v it is 508 KB compiled.

### ISSUE THREAD

**Issue by brizzbane** [vlang/v#5628]
**V version:** V 0.1.28 7004f74.3c432f5
**OS:** Debian 10

**What did you do?**

v -os windows hello_world.v

**What did you expect to see?**

a small executable.  Under 100 KB

**What did you see instead?**

508.0 KiB, hello_world.exe

I'm just checking out the language after seeing it pop up under GitHub trending.  Liked that it supposedly compiled to small binaries and looked like a language that would be much simpler to learn than C or C++.

Am I doing something wrong...or is the bullet point on main page of vlang.io ..incorrect?

**Comment by brizzbane** [comment]
(agian newbie), reading more, I added -prod took it down to 360KB. ...still many times what I was hoping for.

**Comment by JalonSolov** [comment]
Lots of factors involved.  Depends on which version of which C compiler, whether compiling native or cross (like you're doing), etc.

For example, on Ubuntu, using gcc 9.3.0, I see this:
```
jalon@Ripper:~/git/v/examples$ ls -l hello_world*
-rwxr-xr-x 1 jalon jalon 181736 Jul  2 16:59 hello_world
-rwxr-xr-x 1 jalon jalon 489018 Jul  2 16:59 hello_world.exe
-rw-r--r-- 1 jalon jalon     25 Jun 30 15:39 hello_world.v
jalon@Ripper:~/git/v/examples$
```
The version without .exe is Linux version.  Some of that is debug info, so compile both again with `-prod` option, and get this:
```
jalon@Ripper:~/git/v/examples$ ls -l hello_world*
-rwxr-xr-x 1 jalon jalon  25984 Jul  2 17:04 hello_world
-rwxr-xr-x 1 jalon jalon 339328 Jul  2 17:00 hello_world.exe
-rw-r--r-- 1 jalon jalon     25 Jun 30 15:39 hello_world.v
jalon@Ripper:~/git/v/examples$
```
Looks better, but still not as good as it could be.  Now try compiling the same file _on_ windows, and see
```
-a---            7/2/2020  5:05 PM         215147 hello_world.exe
```
with debug, and
```
-a---            7/2/2020  5:11 PM          73137 hello_world.exe
```
without.  That basically tells me that cross-compiling for Windows doesn't do the most efficient code, and neither does a GCC port running on Windows.  :-\

Of course, if you really want to be impressed, try this (only creates an ELF file for now):
```
v -x64 hello_world.v
```
-rwxrwxr-x 1 jalon jalon    188 Jul  2 17:13 hello_world
```
Now, for the last, to compare apples to apples.... if you compile the vweb example, using -prod, you get this:
```
-rwxr-xr-x 1 jalon jalon 61808 Jul  2 17:15 vweb
```
which is well below the 65k mentioned on the web site for a webserver.

So no, there is no bug.  It just depends on what you compile, for which machine, and which thing you used for the backend.

**Comment by medvednikov** [comment]
Good overview @JalonSolov 

And yes, in 99% of cases V web services are going to be running on Linux :)

**Comment by brizzbane** [comment]
I didn't really mean to report 'bug', that just seemed the best of the two options.. I figured a "Hello_world", should compile well below 50kb? ..i guess I was asking how to make that happen/if it is actually possible.  The main page led me to expect small exe files.

**Comment by JalonSolov** [comment]
The main thing I would look at is the C compiler you have.  It appears that it just isn't doing a very efficient job.  We have tested with several different compilers, and found a fairly wide variance.  For example, tcc (tiny c compiler) compiles blazingly fast, but it doesn't do any optimization at all.  gcc does a pretty good job, but the best output is from clang.

It might be worth experimenting with the options available to you, and see which works the best.

Of course, once the `-x64` backend is more mature (it's fairly limited right now), there's no C compiler that will be able to equal it (as evidenced by the 188 byte hello_world I listed).  :-)
